met-web/tests/unit/components/engagementForm.test.tsx

```diff
@@ -1,15 +1,142 @@
+
+    beforeEach(() => {
+        setupEnv();
+    });
+    afterEach(() => {
+        useSelectorMock.mockClear();
+        useDispatchMock.mockClear();
+        useNavigateMock.mockClear();
+        useParamsMock.mockClear();
+        getEngagementMock.mockClear();
+        putEngagementMock.mockClear();
+    });
+
+    test('Test Engagement Form renders and has empty inputs by default', async () => {
+        useParamsMock.mockReturnValue({ engagementId: 'create' });
+        const { getByText } = render(<EngagementForm />);
+        await waitFor(() => {
+            expect(getByText('Engagement Name')).toBeInTheDocument();
+        });
+        expect(screen.getByText('Create Engagement Draft')).toBeInTheDocument();
+    });
+
+    test('Test cannot create engagement with empty fields', async () => {
+        useParamsMock.mockReturnValue({ engagementId: 'create' });
+        const { container, getByText } = render(<EngagementForm />);
+
+        const createButton = getByText('Create Engagement Draft');
+        fireEvent.click(createButton);
+
+        expect(container.querySelectorAll('.Mui-error').length).toBeGreaterThan(0);
+    });
+
+    test('Engagement form with saved engagement should display saved info', async () => {
+        useParamsMock.mockReturnValue({ engagementId: '1' });
+        render(<EngagementForm />);
+
+        await waitFor(() => {
+            expect(screen.getByDisplayValue('Test Engagement')).toBeInTheDocument();
+        });
+
+        expect(getEngagementMock).toHaveBeenCalledOnce();
+        expect(screen.getByText('Update Engagement')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('2022-09-01')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('2022-09-30')).toBeInTheDocument();
+        expect(screen.getByText('Survey 1')).toBeInTheDocument();
+    });
+
+    test('Update engagement button should trigger Put call', async () => {
+        useParamsMock.mockReturnValue({ engagementId: '1' });
+        render(<EngagementForm />);
+
+        await waitFor(() => {
+            expect(screen.getByDisplayValue('Test Engagement')).toBeInTheDocument();
+        });
+        const updateButton = screen.getByText('Update Engagement');
+
+        fireEvent.click(updateButton);
+
+        await waitFor(() => {
+            expect(putEngagementMock).toHaveBeenCalledOnce();
+        });
+    });
+
+    test('Cannot add survey to unsaved engagement', async () => {
+        const openNotificationMock = jest.spyOn(notificationSlice, 'openNotification').mockImplementation(jest.fn());
+        useParamsMock.mockReturnValue({ engagementId: 'create' });
+        render(<EngagementForm />);
+
+        const addSurveyButton = screen.getByText('Add Survey');
+
+        fireEvent.click(addSurveyButton);
+
+        expect(openNotificationMock).toHaveBeenNthCalledWith(1, {
+            severity: 'error',
+            text: 'Please save the engagement before adding a survey',
```

Comment:
Like these too, the test shouldn’t care about the message but that there is an error notification and that’s it

Reply:
Point noted and awaiting input from @saravanpa-aot as well